tests/topotests/bgp_vrf_dynamic_route_leak/test_bgp_vrf_dynamic_route_leak_topo1.py

```diff
@@ -0,0 +1,1856 @@
+        }
+
+        result = verify_bgp_rib(tgen, addr_type, "r1", input_routes_r1)
+        assert result is True, \
+            "Testcase {} : Failed \n Error {}". \
+                format(tc_name, result)
+
+    for addr_type in ADDR_TYPES:
+
+        input_dict_isr={}
+        DUT = ["r1"]
+        VRFS = ["default"]
+        AS_NUM = [100]
+
+        for dut, vrf, as_num in zip(DUT, VRFS, AS_NUM):
+            temp = {dut: {"bgp": []}}
+            input_dict_isr.update(temp)
+
+            temp[dut]["bgp"].append(
+                {
+                    "local_as": as_num,
+                    "vrf": vrf,
+                    "address_family": {
+                        addr_type: {
+                            "unicast": {
+                                "import": {
+                                    "vrf": "ISR"
+                                }
+                            }
+                        }
+                    }
+                })
+
+            temp[dut]["bgp"].append(
+                {
+                    "local_as": as_num,
+                    "vrf": vrf,
+                    "address_family": {
+                        addr_type: {
+                            "unicast": {
+                                "import": {
+                                    "vrf": "route-map rmap_IMP_{}".format(addr_type)
+                                }
+                            }
+                        }
+                    }
+                })
+
+        result = create_router_bgp(tgen, topo, input_dict_isr)
+        assert result is True, "Testcase {} : Failed \n Error: {}". \
+            format(tc_name, result)
+
+    for addr_type in ADDR_TYPES:
+
+        input_routes_r1 = {
+            "r1": {
+                "static_routes": [{
+                    "network": [
+                        NETWORK1_1[addr_type]
+                    ],
+                    "vrf": "default"
+                }]
+            }
+        }
+
+        result = verify_bgp_rib(tgen, addr_type, "r1", input_routes_r1)
+        assert result is True, \
+            "Testcase {} : Failed \n Error {}". \
+                format(tc_name, result)
+
+    for addr_type in ADDR_TYPES:
+
+        step("Delete/Re-add prefix-list ABC.")
+
+        input_dict_pf = {
+            "r1": {
+                "prefix_lists": {
+                    addr_type: {
+                        "pflist_ABC_{}".format(addr_type): [{
+                            "seqid": 10,
+                            "network": NETWORK1_1[addr_type],
+                            "action": "permit",
+                            "delete": True
+                        }]
+                    }
+                }
+            }
+        }
+        result = create_prefix_lists(tgen, input_dict_pf)
+        assert result is True, "Testcase {} : Failed \n Error: {}".format(
+            tc_name, result)
+
+        step("Waiting for 10 sec after deleting prefix-list")
+        sleep(10)
```

Comment:
Are these `sleep`s after add/removing each prefix-list/route-map really necessary? Isn't there a better way to do this?

Suggestions:
- How about asking BGP to go to a known state?
- Are there any log messages that could give topotest a clue that state changed?

Reply:
Hi Rafael, I made some changes in code, which helps to avoid sleep.